nix/docker.nix

```diff
@@ -1,159 +1,179 @@
-#     -e PGPASSFILE=/pgpass
-#     inputoutput/cardano-db-sync run \
-#      --config /configuration/configuration.yaml \
-#      --socket-path /node-ipc/node.socket \
-#      --state-dir /data
-#
-#   * --schema-dir is set within the script
-#
-#  To download and restore a snapshot include -e RESTORE_SNAPSHOT=https://update-cardano-mainnet.iohk.io/cardano-db-sync/db-sync-snapshot-schema-10-block-6014140-x86_64.tgz
-#  See the latest releases for a recent snapshot https://github.com/input-output-hk/cardano-db-sync/releases
-#  See the docker-compose.yml for demonstration of using Docker secrets instead of mounting a pgpass
-#
-#
-############################################################################
-
-{ cardanoLib, dockerTools
-
-# The main contents of the image.
-, cardano-cli, cardano-db-sync, scripts
-
-# Get the current commit
-, gitrev
-
-# Other things to include in the image.
-, bashInteractive, cacert, coreutils, curl, findutils, getconf, glibcLocales
-, gnutar, gzip, jq, iana-etc, iproute, iputils, socat, utillinux, writeScript
-, writeScriptBin, runCommand, runtimeShell, lib, libidn, libpqxx, postgresql
-
-, dbSyncRepoName ? "inputoutput/cardano-db-sync" }:
+{ # Package set passed to NixOS module
+  pkgs
 
-let
+# Tools for building images
+, buildEnv, dockerTools, runCommand, runtimeShell, writeScript, writeScriptBin
+, evalService, cardanoLib
+
+# Image dependencies
+, bashInteractive, cacert, cardano-cli, cardano-db-sync, coreutils, curl, findutils
+, getconf, glibcLocales, gnutar, gzip, jq, iana-etc, iproute, iputils, lib, libidn
+, libpqxx, postgresql, socat, utillinux
+}:
 
+let
   env-shim = runCommand "env-shim" { } ''
     mkdir -p $out/usr/bin
     ln -s ${coreutils}/bin/env $out/usr/bin/env
   '';
 
-  # Layer of tools which aren't going to change much between versions.
   baseImage = dockerTools.buildImage {
-    name = "base-env";
+    name = "cardano-db-sync-base-env";
     config.Env = [ "NIX_SSL_CERT_FILE=${cacert}/etc/ssl/certs/ca-bundle.crt" ];
-    contents = [
-      bashInteractive # Provide the BASH shell
-      cacert # X.509 certificates of public CA's
-      coreutils # Basic utilities expected in GNU OS's
-      curl # CLI tool for transferring files via URLs
-      env-shim # Make /usr/bin/env available
-      findutils # GNU find
-      getconf # get num cpus
-      gnutar # GNU tar
-      glibcLocales # Locale information for the GNU C Library
-      gzip # Gnuzip
-      jq # JSON processor
-      iana-etc # IANA protocol and port number assignments
-      iproute # Utilities for controlling TCP/IP networking
-      iputils # Useful utilities for Linux networking
-      libidn # Library for internationalized domain names
-      libpqxx # A C++ library to access PostgreSQL databases
-      postgresql # A powerful, open source object-relational database system
-      socat # Utility for bidirectional data transfer
-      utillinux # System utilities for Linux
-      cardano-cli
```

Comment:
I actually like these comments here quite a lot. Was there a reason fro dropping them @sgillespie?

Reply:
I thought the majority of them were superfluous. eg, I think it's obvious that `gnutar` provides GNU tar. I don't have a problem adding them back

Reply:
I agree in the case of gnu tar, that that's not _that_ helpful. Same for jq, postgreql, but the other might be a bit trickier to know. Having that documentation isn't bad. If it would say _why_ we need them it would probably be even better, but let's not review this PR to death 🙄